### PR TITLE
[RFC] Fix issue when Enum doesn't display UInt8 correctly.

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Hi,

This PR solves the issue #27025 

As suggested, I've changed the call `print` to `show` when the enum type is `UInt8`.

Regards

### CODE REVIEW

File: base/Enums.jl [hunk 1]
```diff
@@ -162,7 +162,12 @@ macro enum(T, syms...)
             Base.show_datatype(io, t)
             print(io, ":")
             for (sym, i) in $vals
-                print(io, "\n", sym, " = ", i)
+                print(io, "\n", sym, " = ")
+                if (isa(i, UInt8))
```
Comment (nalimilan) [inline base/Enums.jl]: Please use `show` all the time. There's no reason to special-case `UInt8`.

File: test/enums.jl [hunk 2]
```diff
@@ -138,6 +138,10 @@ end
 @test repr(apple) == "apple::$(string(Fruit)) = 0"
 @test string(apple) == "apple"
 
+@enum UI8::UInt8 ten=0x0A el=0x03 t=0x07
```
Comment (nalimilan) [inline test/enums.jl]: Can you move the new tests to line 155, i.e. below existing ones? Here they're in the middle of the  `Fruit` enum tests.

Comment (guilhermeleobas) [reply test/enums.jl]: Done!

**Review by nalimilan — APPROVED** [review]
Thanks!

**Pull request merged**